common/lib/dependabot/logger.rb

```diff
@@ -3,11 +3,17 @@
 require "logger"
 
 module Dependabot
-  def self.logger
-    @logger ||= Logger.new(nil)
-  end
+  module Logger
+    def logger
+      Dependabot::Logger.logger
```

Comment:
@jurre thoughts on this pattern? Feels a bit weird

Reply:
Yeah that makes sense 👍  I think we could simplify even further by removing `def logger` and adding a `module_function :logger` below `def self.logger`.

Reply:
getting a exception trying that `undefined method `logger'` - seems you have to do `def logger ... module_function :logger` but that doesn't seem to work when setting the instance variable from the setter, always uses the nil logger